src/labkey/Query.ts

```diff
@@ -57,4 +60,36 @@ export {
     URL_COLUMN_PREFIX,
     validateQuery,
     Visualization
+}
+
+export interface GetQueryMetadataOptions {
+    containerPath?: string
+    queryName?: string
+    schemaName?: string
+    failure?: (error?: any) => any
+    success?: (data?: any) => any
+}
+
+/**
+ * Gets QueryMetadata in the shape of a Domain design.
+ */
+export function getQueryMetadata(config: GetQueryMetadataOptions): void {
+    let options: GetQueryMetadataOptions = arguments.length > 1 ? {
```

Comment:
I couldn't find a version of this in Query.js. Assuming this is a new endpoint we don't need to have "backwards compatibility" code in here to match arguments. The `config` itself will suffice. Let's remove the `arguments` parsing and rename `config` to `options`.